kratos/utilities/tessellation_utilities/curve_tessellation.h

```diff
@@ -254,23 +307,86 @@ class NurbsCurveTessellation
         return points;
     }
 
-    /**
-    * @brief This method returns the already computed tessellation of a curve
-    * @return return std::vector<std::pair<double, Vector>> tessellation
-    */
+    /* @brief This method iterates through all points within the provided
+     *        polygon and returns the closest point.
+     *
+     * @param CoordinatesArrayType external point
+     * @param rClosestPointGlobalCoordinates closest point within polygon in global coordinates.
+     * @param rClosestPointLocalCoordinates closest point within polygon in local coordinates.
+     * @param rTesselation corresponding tessellation.
+     */
+    static void GetClosestPoint(
+        const CoordinatesArrayType& rPointGlobalCoordinates,
+        CoordinatesArrayType& rClosestPointGlobalCoordinates,
+        CoordinatesArrayType& rClosestPointLocalCoordinates,
+        const TessellationType& rTesselation)
+    {
+        double distance = std::numeric_limits<double>::max();
+        double new_distance;
+        for (IndexType i = 0; i < rTesselation.size(); ++i)
+        {
+            new_distance = norm_2(rPointGlobalCoordinates - std::get<1>(rTesselation[i]));
+            if (new_distance < distance)
+            {
+                distance = new_distance;
+                rClosestPointGlobalCoordinates = std::get<1>(rTesselation[i]);
+                rClosestPointLocalCoordinates[0] = std::get<0>(rTesselation[i]);
+            }
+        }
+    }
+
+    /* @brief This method iterates through all points within the
+     *        polygon and returns the closest point.
+     *
+     * @param CoordinatesArrayType external point
+     * @param rClosestPointGlobalCoordinates closest point within polygon in global coordinates.
+     * @param rClosestPointLocalCoordinates closest point within polygon in local coordinates.
+     * @param rTesselation corresponding tessellation.
+     */
+    void GetClosestPoint(
+        const CoordinatesArrayType& rPointGlobalCoordinates,
+        CoordinatesArrayType& rClosestPointGlobalCoordinates,
+        CoordinatesArrayType& rClosestPointLocalCoordinates) const
+    {
+        GetClosestPoint(
```

Comment:
If this function is const, shuldn't the one that is called here also be const?

Reply:
the other function is static, which is implying const. Thusly, it is running with this.